### PR TITLE
Update embedding.md

### PR DESCRIPTION
**Pull request by Anmol6**

### CODE REVIEW

**Comment by Anmol6** [comment]
I think it's a really nice use case to introduce people to Modal & embedding inference. I do think that the clarity can probably be improved a bit, especially around the core orchestration decisions. 
eg. 
- clarifying we'll be establishing two separate servers — one inference server & one sevrer that just handles the download job.
- what is an inference server in the context of embeddings?

**Comment by jxnl** [comment]
cc: @ivanleomk

File: docs/embedding.md [hunk 1]
```diff
@@ -110,7 +110,7 @@ modal run download.py
 
 ### Text Embedding Inference
 
-In order to embed our function, we'll be using the Hugging Face [Text Embedding Inference](https://github.com/huggingface/text-embeddings-inference) package. We'll walk you through how to leverage caching of model weights by defining a custom modal image, manage container state through a Modal `cls` object methods and lastly, how to leverage this new container in our other functions.
+For our embed our function, we'll be using the Hugging Face [Text Embedding Inference](https://github.com/huggingface/text-embeddings-inference) package. We'll walk through how to leverage caching of model weights by defining a custom modal image, manage container state through a Modal `cls` object methods and lastly, how to leverage this new container in our other functions.
```
Comment (ivanleomk) [inline docs/embedding.md]: Perhaps a rephrase as `For our embedding function` might work better!

File: docs/embedding.md [hunk 2]
```diff
@@ -179,15 +179,15 @@ By using the `run_function` property, we're able to execute the `download_model`
...
 - What to do once the container is shut down using the `__exit__` 
 
-More specifically, in our case, we spawn a server once when the container boots up. This state is then preserved in preparation for future requests that other functions might make so that we only incur the cost of initialising a server once. 
+In our case, we spawn a server once when the container boots up. This state is then preserved in preparation for future requests that other functions might make so that we only incur the cost of initialising a server once. 
 
-All the guesswork of managing the life cycle is taken out of the equation for you with Modal by just defining two functions and using a single decorator. Not only so, we can configure our object to run with a specific image and an attached GPU by modifying the `stub.cls` parameters
+All the guesswork of managing the life cycle is taken out of the equation for you with Modal by just defining two functions and using a single decorator. Not only so, we can also configure our stub object to run with a specific image and an attached GPU by modifying the `stub.cls` parameters
```
Comment (ivanleomk) [inline docs/embedding.md]: Hmm I think the stub object is slightly different from the `stub.cls` object. In this case, we're trying to get the `stub.cls` object to run with different parameters.

In my head

stub => main unit of organisation 

stub.cls => a way to create a definition for an object that is declared on the stub that we can use to get more fine grained control over the lifecycle of the container that will run our function code.

I think we can definitely make it clearer what the `object` I am refering to here is though!

File: docs/embedding.md [hunk 3]
```diff
@@ -289,15 +294,15 @@ def embed_dataset():
   sample_size = int(ttl_size * 0.01) # We work with 1% of the dataset 
   subset = dataset["train"].select(range(sample_size))
   
-	text_chunks = generate_chunks_from_dataset(subset, chunk_size=512)
+  text_chunks = generate_chunks_from_dataset(subset, chunk_size=512)
```
Comment (ivanleomk) [inline docs/embedding.md:297]: Nice catch on the spacing here. Think there's a similar spacing isue for the `dataset=load_from_disk` spacing issue too

File: docs/embedding.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ Before we dive into the code, let's take a look at some of the key concepts that
 
 ### Stubs
 
-A Stub is the most basic concept in Modal. We can declare it using just 2 lines of code.
+A Stub is the most basic concept in Modal — it's a description of how to create a Modal application. We can declare it using just 2 lines of code.
```
Comment (ivanleomk) [inline docs/embedding.md:30]: This is a nice way to put it and is consistent with the modal docs too for a stub - https://modal.com/docs/reference/modal.Stub. It ties in nicely with the subsequently portion where we show how to declare stuff like docker containers, secrets etc

**Review by ivanleomk — CHANGES_REQUESTED** [review]
left some comments on the suggested changes, just some small changes that I think would be good :)

**Comment by jxnl** [comment]
@ivanleomk merge whenever you're feeling good

**Comment by Anmol6** [comment]
@ivanleomk FF to further edit the PR too!

**Review by ivanleomk — APPROVED** [review]
Looks good to me @Anmol6, merging in now.

**Pull request merged**